apps/web/pages/signup.tsx

```diff
@@ -349,197 +353,177 @@ export default function Signup({
+                {/* Upper Row */}
+                <div className="mt-6 flex flex-col gap-2 md:flex-row">
+                  {true ? (
+                    <Button
+                      color="primary"
+                      loading={isGoogleLoading}
+                      CustomStartIcon={
+                        <img
+                          className={classNames(
+                            "text-subtle  mr-2 h-4 w-4 dark:invert",
+                            premiumUsername && "opacity-50"
+                          )}
+                          src="/google-icon-colored.svg"
+                          alt=""
+                        />
+                      }
+                      className={classNames(
+                        "w-full justify-center rounded-md text-center",
+                        formMethods.formState.errors.username ? "opacity-50" : ""
+                      )}
+                      onClick={async () => {
+                        setIsGoogleLoading(true);
+                        const username = formMethods.getValues("username");
+                        const baseUrl = process.env.NEXT_PUBLIC_WEBAPP_URL;
+                        const GOOGLE_AUTH_URL = `${baseUrl}/auth/sso/google`;
+                        const searchQueryParams = new URLSearchParams();
+                        if (username) {
+                          // If username is present we save it in query params to check for premium
+                          searchQueryParams.set("username", username);
+                          localStorage.setItem("username", username);
+                        }
+                        if (token) {
+                          searchQueryParams.set("email", prepopulateFormValues?.email);
+                        }
+                        const url = searchQueryParams.toString()
+                          ? `${GOOGLE_AUTH_URL}?${searchQueryParams.toString()}`
+                          : GOOGLE_AUTH_URL;
+
+                        router.push(url);
+                      }}>
+                      Continue with Google
+                    </Button>
+                  ) : null}
+                </div>
+
                 <div className="mt-6">
                   <div className="relative flex items-center">
                     <div className="border-subtle flex-grow border-t" />
                     <span className="text-subtle mx-2 flex-shrink text-sm font-normal leading-none">
-                      {t("or_continue_with")}
+                      {t("or").toLocaleLowerCase()}
                     </span>
                     <div className="border-subtle flex-grow border-t" />
                   </div>
                 </div>
-              )}
-              <div className="mt-6 flex flex-col gap-2 md:flex-row">
-                {isGoogleLoginEnabled ? (
-                  <Button
-                    color="secondary"
-                    disabled={!!formMethods.formState.errors.username || premiumUsername}
```

Comment:
I removed this because this will look buggy now that the email form and google sign-in button are not on the same page. Lmk what you think cc @sean-brydon

Reply:
Seems fine to me!